### PR TITLE
Perk: Added `TTTCountdownPerk`, `Radar` and fixed TTTPerk issues (`OnEquip()`, ...)

### PR DESCRIPTION
**Pull request by Alf21**
- Added `TTTPerk > Radar`
- Fixed multiple `OnEquip` calls
- Added `IItem.Simulate`

image

### CODE REVIEW

File: code/items/perks/Test1.cs [hunk 1]
```diff
@@ -9,6 +11,33 @@ public Test1() : base()
         {
 
         }
+        public override void OnEquip()
+        {
+            base.OnEquip();
+
+            VertexBuffer vb = new VertexBuffer(); // this should be called vertex builder or something
```
Comment (mzegar) [inline code/items/perks/Test1.cs:18]: Is this comment needed, `VertexBuffer` is within s&box right? Not a change we can make.

File: code/items/perks/Test1.cs [hunk 2]
```diff
@@ -9,6 +11,33 @@ public Test1() : base()
...
+
+            AnimEntity entity = new AnimEntity();
+            entity.SetModel(model);
+            entity.SetupPhysicsFromModel(PhysicsMotionType.Static);
+            entity.GlowActive = true;
+            entity.GlowColor = Color.Red;
+            entity.GlowState = GlowStates.GlowStateOn;
+            entity.GlowDistanceStart = 1;
+            entity.GlowDistanceEnd = 10000;
+            entity.Position = Owner.Position + Owner.EyeRot.Forward * 100f;
```
Comment (mzegar) [inline code/items/perks/Test1.cs:39]: If you find any of these values useful to define as constants, please do,

ex. `GLOW_DISTANCE_START = 1` // probably a better name

Just whenever someone needs to either modify these values in the future, or re-write them.

File: code/items/perks/radar/Radar.cs [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+
+using Sandbox;
+
+using TTTReborn.Player;
+using TTTReborn.UI;
+
+namespace TTTReborn.Items
+{
+    [Library("ttt_radar")]
+    public partial class Radar : TTTCountdownPerk, IBuyableItem
```
Comment (mzegar) [inline code/items/perks/radar/Radar.cs:12]: This radar stuff is neat.

File: code/items/perks/radar/RadarPoint.cs [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+            }
+
+            DistanceLabel.Text = $"{player.Position.Distance(Position):n0}";
+
+            Vector3 screenPos = Position.ToScreen();
+            bool visible = screenPos.z > 0f;
+
+            SetClass("hide", !visible);
+
+            // visible
```
Comment (mzegar) [inline code/items/perks/radar/RadarPoint.cs:43]: No need for this comment, variable explains it well enough :)

**Review by mzegar — COMMENTED** [review]
Got some comments

File: code/items/perks/Test1.cs (hunk 2 above)
Comment (Alf21) [reply code/items/perks/Test1.cs:39]: Test1 is just a test for ingame things, nothing that has any function or should be kept, so please ignore this :D

File: code/items/perks/radar/RadarPoint.cs (hunk 4 above)
Comment (Alf21) [reply code/items/perks/radar/RadarPoint.cs:43]: Haha yea :D

**Review by mzegar — APPROVED** [review]
LGTM, nice stuff.

**Pull request merged**